src/renderer/fileListView/fileListHeader.tsx

```diff
@@ -10,27 +10,32 @@ import ChevronArrowSVG from '../_assets/svg/chevron-arrow-right.svg';
 import NavigationPrompt from 'react-router-navigation-prompt';
 import { WarningModal } from './WarningModal';
 import { ConfirmOptimizationModal } from './ConfirmOptimizationModal';
+import { store, selectors } from '../store';
 
 interface IFileListHeaderProps {
 	siteImageData: SiteImageData,
 	setOverviewSelected: (x: boolean) => void,
     invokeModal: (ModalContents: React.FC, onCancel?: Function, onConfirm?: Function) => void,
-	getAllChecked: () => ImageData[],
-	onCancel: () => void,
+	selectedImages: ImageData[],
+	cancelImageCompression: () => void,
 	resetToOverview: () => void,
+	siteID: string,
 }
 
 export const FileListHeader = (props: IFileListHeaderProps) => {
 	const {
 		siteImageData,
 		setOverviewSelected,
 		invokeModal,
-		getAllChecked,
-		onCancel,
+		selectedImages,
+		cancelImageCompression,
 		resetToOverview,
 	} = props;
 
-    const disableButton = getAllChecked().length > 0 ? false : true;
+	const compressedImagesOriginalSize = selectors.originalSizeOfCompressedImages(store.getState(), props);
```

Comment:
Agree with avoiding the `store.getState()`; in fact this should be avoided 100% of the time in a component. Delegate this to the selector implementation. It will remove the need to refactor down the road by allowing selectors to reuse other selectors. Lastly by getting the state within a component you invite mutation to said state, which creates bugs that are really hard to fix.

In regards to saving props values, I probably don't fully understand, but I think this is a bit dangerous as this might remove memoization that Reselect offers.

Reply:
Ha! Same thought on my end. I'm going to try and export some bound versions of these bad bois and clean up the way that `siteID` is passed around